pkg/sql/opt/exec/execbuilder/testdata/select_for_update

```diff
@@ -525,78 +525,40 @@ root                 ·                 ·
 query TTT
 EXPLAIN SELECT * FROM t WHERE a IN (SELECT a FROM t) FOR UPDATE
 ----
-·           distribution        local
-·           vectorized          true
-merge-join  ·                   ·
- │          type                semi
- │          equality            (a) = (a)
- │          left cols are key   ·
- │          right cols are key  ·
- │          mergeJoinOrder      +"(a=a)"
- ├── scan   ·                   ·
- │          table               t@primary
- │          spans               FULL SCAN
- │          locking strength    for update
- └── scan   ·                   ·
-·           table               t@primary
-·           spans               FULL SCAN
+·     distribution      local
```

Comment:
I'm not sure why these changed, my changes shouldn't have affected the plans like this. Though the plan does seem reasonable since the in is a tautology?